server/eclipse-project/pom.xml

```diff
@@ -4,7 +4,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>us.freeandfair.production</groupId>
 	<artifactId>colorado_rla</artifactId>
-	<version>1.0.3</version>
+	<version>1.0.4</version>
```

Comment:
I vote to have this say something like 1.0.4.dev1 until it is merged into master. My test scripts now incorporate version numbers better into the metadata saved with each test, and this will clarify what was actually tested.
I'd actually prefer 1.0.4.dev-g1234567 with a commit id in there, but that would take some clever scripting work.
FWIW, these version number formats match the guidelines at https://docs.openstack.org/pbr/latest/user/semver.html  which is a take-off on semantic versioning 2.0, with Python PEP-440 compatibility.

Reply:
I don't really want to do the clever scripting work, and having to commit a new commit after the PR is approved to change the version number from 1.0.4-dev to 1.0.4 is just asking for that step to be forgotten, repeatedly. But I'll see if I can think of something.

Reply:
I figure we just need a checklist, and an extra minute, to accompany use of existing and/or improved scripts for doing a distribution (ala `script/build-release`). For now I'm also not volunteering to make a more clever script, so I'm just suggesting that e.g. for each release, at whatever time we decide "ship it", we 

* do a commit with the full release number 1.0.4
* build it
* tag it
* do a commit with the next release number 1.0.5.dev1.

Not much to forget then.
And when we do get around to doing a script, it can follow the PBR practice (documented along with the link up above) of actually just pulling the version from the tag itself, so no extra merges are necessary. That will be a small but helpful incentive to find or write one like that....

Reply:
Short version: I like @nealmcb's second suggestion, handling everything in one PR at the version boundary. I would only explicitly add that we needn't add a number to the `dev`-like suffix (e.g. just `1.0.5-dev` or `1.0.5-alpha` is fine). In any case, I don't think git hashes should be included in the version numbers.

Reply:
(I see why @nealmcb might want `dev`/`alpha` suffix to be numbered, but IMO it would _ideal_ if that sort of tracking were managed at the level of the test scripts, which I'd guess would either know exactly what commit they are dealing with, or they are dealing with a release candidate on a test server, which I would consider a "test release", and thus its own natural tagging boundary).

Reply:
I’ll add a “-dev” on it.